Goal: terminate argument validation in login when it hits a '--'.
Fixes: #66368

Status wrt upstream: It could certainly be submitted to upstream.

Index: shadow-4.0.3/src/login.c
===================================================================
--- shadow-4.0.3.orig/src/login.c	2005-05-23 01:23:54.377167000 +0200
+++ shadow-4.0.3/src/login.c	2005-05-23 01:24:36.427167000 +0200
@@ -294,6 +294,8 @@
 	for (arg = 1; arg < argc; arg++) {
 		if (argv[arg][0] == '-' && strlen (argv[arg]) > 2)
 			usage ();
+		if (!strcmp(argv[arg], "--"))
+			break; /* stop checking on a "--" */
 	}
 }
 
